sstables/compaction.cc

```diff
@@ -512,9 +520,25 @@ class compaction {
         report_start(formatted_msg);
 
         _compacting = std::move(ssts);
-        return make_sstable_reader();
+
+        auto now = gc_clock::now();
+        auto consumer = make_interposer_consumer([this, gc_consumer = std::move(gc_consumer), now] (flat_mutation_reader reader) mutable
+        {
+            using compact_mutations = compact_for_compaction<compacting_sstable_writer, GCConsumer>;
+            auto cfc = make_stable_flattened_mutations_consumer<compact_mutations>(*schema(), now,
```

Comment:
Why do you need a stable flattened mutation consumer? Since you're running in a thread, you could create it there and it would be stable (by virtue of sitting on a thread stack). This saves on indirection.

Reply:
First, this code was already like this inside `run()`. I moved here with as few changes as possible.

Second, I don't think this is running in a thread. The consumer object is created in a thread but I don't think it is necessarily guaranteed to be called in a thread.

Which is why I had to create one below. I can definitely move the creation to inside the thread, no reason not to if that's what you prefer. It'd just be more different from the original code in this pass.

Reply:
If it's not changed, best to keep it unchanged and make optimizations (if possible) separately.